tests/unit/loader/test_tf_dataloader.py

```diff
@@ -369,37 +368,36 @@ def test_mh_support(tmpdir, batch_size):
         batch_size=batch_size,
         shuffle=False,
     )
-    nnzs = None
     idx = 0
+
+    multihot_data = df
+    if HAS_GPU:
+        multihot_data = multihot_data.to_pandas()
+
     for X, y in data_itr:
-        assert len(X) == 4
+        assert len(X) == 7
         n_samples = y.shape[0]
 
         for mh_name in ["Authors", "Reviewers", "Embedding"]:
             # assert (mh_name) in X
-            array, nnzs = X[mh_name]
-            nnzs = nnzs.numpy()[:, 0]
-            array = array.numpy()[:, 0]
-
-            if mh_name == "Embedding":
-                assert (nnzs == 3).all()
-            else:
-                lens = [
-                    len(x) for x in data[mh_name][idx * batch_size : idx * batch_size + n_samples]
-                ]
-                assert (nnzs == np.array(lens)).all()
+            array, offsets = X[f"{mh_name}__values"], X[f"{mh_name}__offsets"]
```

Comment:
Using `merlin.table.TensorTable` here would simplify this and also make this code work with the previous version of the dataloader too.

```python
mh_col = TensorTable(X)[f"{mh_name}"]
values, offsets = mh_col.values.numpy(), mh_col.offsets.numpy()